rubix-core/src/main/java/com/qubole/rubix/core/CachingInputStream.java

```diff
@@ -90,67 +89,48 @@ public Thread newThread(Runnable r)
   private final int bufferSize;
   BookKeeperFactory bookKeeperFactory;
 
-  public CachingInputStream(FileSystem parentFs, Path backendPath, Configuration conf,
+  public CachingInputStream(Path backendPath, Configuration conf,
                             CachingFileSystemStats statsMbean, ClusterType clusterType,
                             BookKeeperFactory bookKeeperFactory, FileSystem remoteFileSystem,
                             int bufferSize, FileSystem.Statistics statistics) throws IOException
   {
-    initialize(backendPath.toString(), conf, bookKeeperFactory);
-    this.bookKeeperFactory = bookKeeperFactory;
+    this.conf = conf;
+    this.strictMode = CacheConfig.isStrictMode(conf);
     this.remotePath = backendPath.toString();
+    this.localPath = CacheUtil.getLocalPath(remotePath, conf);
+    this.blockSize = CacheConfig.getBlockSize(conf);
+    this.diskReadBufferSize = CacheConfig.getDiskReadBufferSize(conf);
+    this.bookKeeperFactory = bookKeeperFactory;
     this.remoteFileSystem = remoteFileSystem;
 
-    try (RetryingPooledBookkeeperClient bookKeeperClient = bookKeeperFactory.createBookKeeperClient(conf)) {
-      FileInfo fileInfo = bookKeeperClient.getFileInfo(backendPath.toString());
-      this.fileSize = fileInfo.fileSize;
-      this.lastModified = fileInfo.lastModified;
-    }
-    catch (Exception ex) {
-      if (strictMode) {
-        throw Throwables.propagate(ex);
-      }
-      log.error(String.format("Could not get FileInfo for %s. Fetching FileStatus from remote file system :", backendPath.toString()), ex);
-      FileStatus fileStatus = parentFs.getFileStatus(backendPath);
+    if (CacheConfig.isFileStalenessCheckEnabled(conf)) {
+      FileStatus fileStatus = remoteFileSystem.getFileStatus(backendPath);
       this.fileSize = fileStatus.getLen();
       this.lastModified = fileStatus.getModificationTime();
     }
+    else {
```

Comment:
can remove duplication by changing this a bit:
```
fileSize = -1
if (!stalenessChecksEnabled):
    try:
        set fileSize with thrift call
    catch:
        log

if fileSize == -1:
    set fileSize and lastModified via getFileStatus
```

Reply:
Done